src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -134,31 +160,43 @@ class DeltaCatalog(val spark: SparkSession) extends DelegatingCatalogExtension
       schema: StructType,
       partitions: Array[Transform],
       properties: util.Map[String, String]): StagedTable = {
-    throw new IllegalStateException("not supported yet")
+    properties.get("provider") match {
+      case "delta" =>
+        new StagedDeltaTableV2(
+          ident, schema, partitions, properties, TableCreationModes.ReplaceTable)
+      case _ =>
+        super.dropTable(ident)
+        OtherFormatsStagedTable(this, ident, schema, partitions, properties)
+    }
   }
 
   override def stageCreateOrReplace(
       ident: Identifier,
       schema: StructType,
       partitions: Array[Transform],
       properties: util.Map[String, String]): StagedTable = {
-    throw new IllegalStateException("not supported yet")
+    properties.get("provider") match {
+      case "delta" => new StagedDeltaTableV2(
+        ident, schema, partitions, properties, TableCreationModes.CreateOrReplaceTable)
+      case _ =>
+        try super.dropTable(ident) catch {
+          case _: NoSuchTableException | _: NoSuchDatabaseException | _: NoSuchNamespaceException =>
+            // fine, maybe we're just creating
```

Comment:
nit: I was a bit confused here for a second, maybe point out that we're dropping to cover the case where we're replacing